packages/@glimmer/reference/test/references-test.ts

```diff
@@ -37,111 +34,284 @@ class TrackedDict<T> {
-  assert.equal(reference.value(), 'BAR');
+      assert.equal(valueForRef(childRef), 123, 'value is correct');
+      assert.equal(valueForRef(childRef), 123, 'value is correct');
+      assert.equal(getCount, 1, 'get called correct number of times');
 
-  assert.strictEqual(computed, 2, 'computed');
+      parent.child = 456;
 
-  dict.set('baz', 'bat');
+      assert.equal(valueForRef(childRef), 456, 'value updated correctly');
+      assert.equal(valueForRef(childRef), 456, 'value is correct');
+      assert.equal(getCount, 2, 'get called correct number of times');
 
-  assert.equal(reference.value(), 'BAR');
-  assert.equal(reference.value(), 'BAR');
-  assert.equal(reference.value(), 'BAR');
+      assert.equal(isUpdatableRef(childRef), true, 'childRef is updatable');
 
-  assert.strictEqual(computed, 3, 'computed');
+      updateRef(childRef, 789);
 
-  dict.set('foo', 'bar');
+      assert.equal(valueForRef(childRef), 789, 'value updated correctly');
+      assert.equal(getCount, 3, 'get called correct number of times');
+      assert.equal(setCount, 1, 'set called correct number of times');
+    });
+  });
 
-  assert.equal(reference.value(), 'bar');
-  assert.equal(reference.value(), 'bar');
-  assert.equal(reference.value(), 'bar');
+  module('compute ref', () => {
+    test('compute reference caches computation', assert => {
+      let count = 0;
 
-  assert.strictEqual(computed, 4, 'computed');
-});
+      let dict = new TrackedDict<string>();
+      let ref = createComputeRef(() => {
+        count++;
+        return dict.get('foo');
+      });
 
-QUnit.test('CachedReference caches nested computation correctly', assert => {
-  let computed = 0;
+      dict.set('foo', 'bar');
 
-  class DictValueReference extends CachedReference<string> {
-    constructor(private parent: Reference<TrackedDict<string>>, private key: string) {
-      super();
-    }
+      assert.strictEqual(count, 0, 'precond');
 
-    compute(): string {
-      computed++;
-      return this.parent.value().get(this.key);
-    }
-  }
+      assert.equal(valueForRef(ref), 'bar');
+      assert.equal(valueForRef(ref), 'bar');
+      assert.equal(valueForRef(ref), 'bar');
+
+      assert.strictEqual(count, 1, 'computed');
+
+      dict.set('foo', 'BAR');
+
+      assert.equal(valueForRef(ref), 'BAR');
+      assert.equal(valueForRef(ref), 'BAR');
+      assert.equal(valueForRef(ref), 'BAR');
+
+      assert.strictEqual(count, 2, 'computed');
+
+      dict.set('baz', 'bat');
+
+      assert.equal(valueForRef(ref), 'BAR');
+      assert.equal(valueForRef(ref), 'BAR');
+      assert.equal(valueForRef(ref), 'BAR');
+
+      assert.strictEqual(count, 3, 'computed');
+
+      dict.set('foo', 'bar');
+
+      assert.equal(valueForRef(ref), 'bar');
+      assert.equal(valueForRef(ref), 'bar');
+      assert.equal(valueForRef(ref), 'bar');
+
+      assert.strictEqual(count, 4, 'computed');
+    });
+
+    test('CachedReference caches nested computation correctly', assert => {
```

Comment:
I don't think this name is correct any more

```suggestion
    test('compute reference caches nested computation correctly', assert => {
```